### PR TITLE
Remove ansi codes from non step test

### PR DESCRIPTION
**Pull request by NiklasBack**
This PR adds additional ANSI stripping due to amongs other the extra backslashes JSON.stingify adds. It also changes stepCategory and summary to optional.
Because I commited and pushed before Biome check (sorry for that) I have commented those parts in index.ts and xray.types.ts that I have edited,

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -98,29 +98,29 @@ class XrayReporter implements Reporter {
...
       if (result.errors.length > 0 && !pwStepsExists) {
-        xrayTestData.comment = JSON.stringify(result.errors);
+        xrayTestData.comment = this.stripAnsi(JSON.stringify(result.errors).replace(/\\\\/g, "\\"));
       } else {
         await Promise.all(
           result.steps.map(async (step) => {
             if (this.stepCategories.some((type) => type.includes(step.category))) {
               // Add Step to request
-              const errorMessage = this.stripAnsi(step);
+              const errorMessage = this.stripAnsi(step.error?.stack?.valueOf() as string);
```
Comment (NiklasBack) [inline src/index.ts:113]: Modified here

File: src/index.ts [hunk 2]
```diff
@@ -133,51 +133,57 @@ class XrayReporter implements Reporter {
...
+        case "ABORTED":
           console.log(`${bold(white(`🚫 ${projectID}${testCase.title}`))}`);
           break;
       }
     }
   }
 
-  private stripAnsi(step: TestStep) {
-    const ST = '(?:\\u0007|\\u001B\\u005C|\\u009C)';
+  private stripAnsi(step: string) {
```
Comment (NiklasBack) [inline src/index.ts:173]: And this block

File: src/types/xray.types.ts [hunk 3]
```diff
@@ -27,8 +27,8 @@ export interface XrayOptions {
   uploadTrace?: boolean;
   uploadVideo?: boolean;
   markFlakyWith?: string;
-  stepCategories: string[];
-  summary: string;
+  stepCategories?: string[];
```
Comment (NiklasBack) [inline src/types/xray.types.ts:30]: And these two rows

File: src/index.ts [hunk 4]
```diff
@@ -98,29 +98,29 @@ class XrayReporter implements Reporter {
...
-        comment: '',
+        comment: "",
       };
 
       // Set Test Error
-      const pwStepsExists = result.steps.some((step) => step.category.includes('test.step'));
+      const pwStepsExists = result.steps.some((step) => step.category.includes("test.step"));
       if (result.errors.length > 0 && !pwStepsExists) {
-        xrayTestData.comment = JSON.stringify(result.errors);
+        xrayTestData.comment = this.stripAnsi(JSON.stringify(result.errors).replace(/\\\\/g, "\\"));
```
Comment (NiklasBack) [inline src/index.ts:107]: Modified here

**Review by inluxc — APPROVED** [review]

**Pull request merged**